### PR TITLE
HKG CAN-FD: steering pressed filter

### PR DESCRIPTION
**Pull request by sshane**

### CODE REVIEW

File: selfdrive/car/interfaces.py [hunk 1]
```diff
@@ -364,6 +365,11 @@ def update_blinker_from_lamp(self, blinker_time: int, left_blinker_lamp: bool, r
     self.right_blinker_cnt = blinker_time if right_blinker_lamp else max(self.right_blinker_cnt - 1, 0)
     return self.left_blinker_cnt > 0, self.right_blinker_cnt > 0
 
+  def update_steering_pressed(self, steering_pressed, steering_pressed_min_count):
+    self.steering_pressed_cnt += 1 if steering_pressed else -1
+    self.steering_pressed_cnt = clip(steering_pressed_cnt, 0, steering_pressed_min_count)
+    return self.steering_pressed_cnt >= steering_pressed_min_count
```
Comment (sshane) [inline selfdrive/car/interfaces.py:371]:
```suggestion
    self.steering_pressed_cnt = clip(steering_pressed_cnt, 0, steering_pressed_min_count + 1)
    return self.steering_pressed_cnt > steering_pressed_min_count
```

File: selfdrive/car/interfaces.py [hunk 2]
```diff
@@ -364,6 +365,11 @@ def update_blinker_from_lamp(self, blinker_time: int, left_blinker_lamp: bool, r
     self.right_blinker_cnt = blinker_time if right_blinker_lamp else max(self.right_blinker_cnt - 1, 0)
     return self.left_blinker_cnt > 0, self.right_blinker_cnt > 0
 
+  def update_steering_pressed(self, steering_pressed, steering_pressed_min_count):
+    self.steering_pressed_cnt += 1 if steering_pressed else -1
+    self.steering_pressed_cnt = clip(steering_pressed_cnt, 0, steering_pressed_min_count + 1)
```
Comment (sshane) [inline selfdrive/car/interfaces.py:370]:
```suggestion
    self.steering_pressed_cnt = clip(self.steering_pressed_cnt, 0, self.steering_pressed_min_count + 1)
```

File: selfdrive/car/hyundai/carstate.py [hunk 3]
```diff
@@ -188,7 +188,7 @@ def update_canfd(self, cp, cp_cam):
     ret.steeringAngleDeg = cp.vl["STEERING_SENSORS"]["STEERING_ANGLE"] * -1
     ret.steeringTorque = cp.vl["MDPS"]["STEERING_COL_TORQUE"]
     ret.steeringTorqueEps = cp.vl["MDPS"]["STEERING_OUT_TORQUE"]
-    ret.steeringPressed = abs(ret.steeringTorque) > self.params.STEER_THRESHOLD
+    ret.steeringPressed = self.update_steering_pressed(abs(ret.steeringTorque) > self.params.STEER_THRESHOLD, 20)
```
Comment (sshane) [inline selfdrive/car/hyundai/carstate.py:191]:
```suggestion
    ret.steeringPressed = self.update_steering_pressed(abs(ret.steeringTorque) > self.params.STEER_THRESHOLD, 5)
```

File: selfdrive/car/interfaces.py [hunk 4]
```diff
@@ -364,6 +365,11 @@ def update_blinker_from_lamp(self, blinker_time: int, left_blinker_lamp: bool, r
     self.right_blinker_cnt = blinker_time if right_blinker_lamp else max(self.right_blinker_cnt - 1, 0)
     return self.left_blinker_cnt > 0, self.right_blinker_cnt > 0
 
+  def update_steering_pressed(self, steering_pressed, steering_pressed_min_count):
```
Comment (adeebshihadeh) [inline selfdrive/car/interfaces.py:368]: can you add a comment that explains this is for noisy torque signals?

Comment (adeebshihadeh) [reply selfdrive/car/interfaces.py:368]: then let's merge

File: selfdrive/car/interfaces.py [hunk 5]
```diff
@@ -364,6 +365,11 @@ def update_blinker_from_lamp(self, blinker_time: int, left_blinker_lamp: bool, r
...
+    self.steering_pressed_cnt += 1 if steering_pressed else -1
```
Comment (sshane) [inline selfdrive/car/interfaces.py:369]:
```suggestion
  def update_steering_pressed(self, steering_pressed, steering_pressed_min_count):
    """Applies filtering on noisy driver torque signals for steering pressed."""
    self.steering_pressed_cnt += 1 if steering_pressed else -1
```

File: selfdrive/car/interfaces.py [hunk 6]
```diff
@@ -364,6 +365,12 @@ def update_blinker_from_lamp(self, blinker_time: int, left_blinker_lamp: bool, r
     self.right_blinker_cnt = blinker_time if right_blinker_lamp else max(self.right_blinker_cnt - 1, 0)
     return self.left_blinker_cnt > 0, self.right_blinker_cnt > 0
 
+  def update_steering_pressed(self, steering_pressed, steering_pressed_min_count):
+    """Applies filtering on steering pressed for noisy driver torque signals."""
+    self.steering_pressed_cnt += 1 if steering_pressed else -1
+    self.steering_pressed_cnt = clip(self.steering_pressed_cnt, 0, self.steering_pressed_min_count + 1)
```
Comment (sshane) [inline selfdrive/car/interfaces.py:371]:
```suggestion
    self.steering_pressed_cnt = clip(self.steering_pressed_cnt, 0, steering_pressed_min_count + 1)
```

File: selfdrive/car/interfaces.py [hunk 7]
```diff
@@ -364,6 +365,12 @@ def update_blinker_from_lamp(self, blinker_time: int, left_blinker_lamp: bool, r
     self.right_blinker_cnt = blinker_time if right_blinker_lamp else max(self.right_blinker_cnt - 1, 0)
     return self.left_blinker_cnt > 0, self.right_blinker_cnt > 0
 
+  def update_steering_pressed(self, steering_pressed, steering_pressed_min_count):
+    """Applies filtering on steering pressed for noisy driver torque signals."""
+    self.steering_pressed_cnt += 1 if steering_pressed else -1
+    self.steering_pressed_cnt = clip(self.steering_pressed_cnt, 0, steering_pressed_min_count + 1)
```
Comment (sshane) [inline selfdrive/car/interfaces.py:371]:
```suggestion
    self.steering_pressed_cnt = clip(self.steering_pressed_cnt, 0, steering_pressed_min_count * 2)
```

**Pull request merged**